client/js/azure/util.js

```diff
@@ -6,11 +6,55 @@ qq.azure.util = qq.azure.util || (function() {
     return {
         AZURE_PARAM_PREFIX: "x-ms-meta-",
 
+        /** Test if a request header is actually a known Azure parameter.
+         *
+         * @param name Name of the Request Header parameter to construct a (possibly) prefixed name.
+         * @returns {Boolean} Test result.
+         */
+        _paramNameMatchesAzureParameter: function(name) {
+            switch (name)
+            {
+                //
+                // Valid request headers (not sent by fine-uploader) which should be returned as-is (case-sensitive)
+                // see: https://msdn.microsoft.com/en-us/library/azure/dd179451.aspx
+                //
+                case "Cache-Control":
+                case "Content-Disposition":
+                case "Content-Encoding":
+                case "Content-MD5":
+                case "x-ms-blob-content-encoding":
+                case "x-ms-blob-content-disposition":
+                case "x-ms-blob-content-md5":
+                case "x-ms-blob-cache-control":
+                    return true;
+                default:
+                    return false;
+            }
+        },
+
+        /** Create Prefixed request headers which are appropriate for Azure.
+         *
+         * If the request header is appropriate for Azure (e.g. Cache-Control) then it should be
+         * passed along without a metadata prefix. For all other request header parameter names,
+         * qq.azure.util.AZURE_PARAM_PREFIX should be prepended.
+         *
+         * @param name Name of the Request Header parameter to construct a (possibly) prefixed name.
+         * @returns {String} A valid Request Header parameter name.
+         */
+        _getPrefixedParamName: function(name) {
+            if (qq.azure.util._paramNameMatchesAzureParameter(name)) {
+                return name;
+            }
+            else {
+                return qq.azure.util.AZURE_PARAM_PREFIX + name;
+            }
+        },
+
         getParamsAsHeaders: function(params) {
             var headers = {};
 
             qq.each(params, function(name, val) {
-                var headerName = qq.azure.util.AZURE_PARAM_PREFIX + name;
+                var headerName = qq.azure.util._getPrefixedParamName(name);
 
                 if (qq.isFunction(val)) {
```

Comment:
The next scheduled release is probably at least a couple months away. Either way this change can wait, so, not a problem.

On Wed, May 20, 2015 at 3:30 AM, Korijn van Golen
<notifications@github.com> wrote:

>>          getParamsAsHeaders: function(params) {
>>              var headers = {};
>>  
>>              qq.each(params, function(name, val) {
>> -                var headerName = qq.azure.util.AZURE_PARAM_PREFIX + name;
>> +                var headerName = qq.azure.util._getPrefixedParamName(name);
>>  
>>                  if (qq.isFunction(val)) {
> I'm a bit short on time for this final adjustment, can we postpone it till after release?
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/FineUploader/fine-uploader/pull/1404/files#r30680244